packages/path-compare/src/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Comparison function for two values of the same type `T`.
+ *
+ * Can be used with `Array.prototype.sort` and other similar contexts
+ *
+ * @template T The type of the values to compare
+ * @callback CompareFn
+ * @param {T} a First value
+ * @param {T} b Second value
+ * @returns {number} Negative integer if `a < b`; positive integer if `a > b`;
+ * `0` or `NaN` if equal
+ */
+
+const { stringify: q } = JSON;
+
+/**
+ * String comparison function based on UTF-16 code units
+ *
+ * @type {CompareFn<string>}
+ */
+// eslint-disable-next-line no-nested-ternary
+export const stringCompare = (a, b) => (a === b ? 0 : a < b ? -1 : 1);
+
+/**
+ * Reducer to calculate the cumulative length of a string array
+ * @param {number} length Current length
+ * @param {string} value Some string
+ * @returns {number} New length
+ */
+const cumulativeLength = (length, value) => {
+  return length + value.length;
+};
+
+/**
+ * Compares two string arrays and returns a comparison result.
+ *
+ * The algorithm is as follows:
+ *
+ * 1. _Check if either value is `undefined`._ If both are `undefined`, return `0`. If `a` is `undefined`, return `1`. If `b` is `undefined`, return `-1`.
+ * 2. _Check the lengths of the arrays._ If they are different, return the difference.
+ * 3. _Check the cumulative lengths of the arrays_ using the count of UTF-16 units in each string. If they are different, return the difference.
+ * 4. _Check the individual elements of the arrays_ via lexical comparison. If they are different, return the difference.
+ * 5. _If all elements are the same_ ("deep equality"), return `0`.
```

Comment:
```suggestion
 * 1. _Check if either value is `undefined`._ If both are `undefined`, return
      `0`. If `a` is `undefined`, return `1`. If `b` is `undefined`, return
      `-1`.
 * 2. _Check the lengths of the arrays._ If they are different, return the
      difference.
 * 3. _Check the cumulative lengths of the arrays_ using the count of UTF-16
      code units in each string. If they are different, return the difference.
 * 4. _Check the individual elements of the arrays_ via lexical comparison. If
      they are different, return the difference.
 * 5. _If all elements are the same_ ("deep equality"), return `0`.
```

Reply:
formatting changes should be applied by tooling 😄

Reply:
(I did this manually. something like [prettier-plugin-jsdoc](https://www.npmjs.com/package/prettier-plugin-jsdoc) would be helpful)